asv_bench/benchmarks/arithmetic.py

```diff
@@ -469,4 +469,29 @@ def time_apply_index(self, offset):
         offset.apply_index(self.rng)
 
 
+class Sub:
+    def setup(self):
+        date_range = pd.date_range("20200101 00:00", "20200102 0:00", freq="S")
+        level_0_names = [str(i) for i in range(30)]
+
+        index = pd.MultiIndex.from_product([level_0_names, date_range])
+        column_names = ["col_1", "col_2"]
+
+        self.df = pd.DataFrame(
+            np.random.rand(len(index), 2),
+            index=index,
+            columns=column_names
+        )
+
+        self.sub_df = pd.DataFrame(
+            np.random.randint(1, 10, (len(level_0_names), 2)),
+            index=level_0_names,
+            columns=column_names,
+        )
+
+    def subsequent_sub_application(self):
```

Comment:
this needs to be of the form `def time_foo(self)`

Reply:
Thanks, did not know that. In hindsight it is obvious...